src/lib.rs

```diff
@@ -1,121 +1,76 @@
-use std::{
-    fs,
-    fs::File,
-    io,
-    path::{Path, PathBuf},
-};
+use std::{fs::{File}, io, path::{Path, PathBuf}};
+use async_std::{fs::{create_dir_all, set_permissions, Permissions}};
+use zip::read::ZipFile;
 
-pub fn unzip_file(file: File, directory: &Path) {
-    let mut archive = zip::ZipArchive::new(file).unwrap();
+pub async fn unzip_archive(file: File, directory: &Path) -> Result<(), Box<dyn std::error::Error>> {
+    let mut archive = zip::ZipArchive::new(file)?;
+    let mut files = vec![];
```

Comment:
this `Vec<Result<...>>` should still be used and dealt with.

Reply:
I would call this variable "results" or something like that as it contains the results and not the file itself.

Reply:
Renamed.